### PR TITLE
[CodeSmell] Cleanup

### PR DESCRIPTION
**Pull request by kikkiii21**
- removed commented code in SignIn.js removed duplicate import in EditEvent.js
- removed unused imports in EventColorDisplay.js
- removed unused function in EventColorSelector.js
- Removed all minor issues from SonarCloud

### CODE REVIEW

File: src/components/StudyRoom/StudyRoomHome.js [hunk 1]
```diff
@@ -10,26 +10,20 @@ import Typography from "@mui/material/Typography";
...
+    const user = GetAuthentication();
 
     useEffect(() => {
-        //user needs to be logged in to access
-        if (email === undefined || email === '') {
-            navigate("/login");
-        }
         getData();
-    }, [email])
+    })
```
Comment (mnashat) [inline src/components/StudyRoom/StudyRoomHome.js:22]: Did this compile with no dependency array?

File: src/components/Calendar/EventColorSelector/EventColorSelector.js [hunk 2]
```diff
@@ -12,14 +12,13 @@ import MenuItem from "@mui/material/MenuItem";
...
 import { useNavigate } from "react-router";
-import { useState } from "react";
 import PersistentDrawerLeft from '../../NavDrawer/navDrawer'
 import { CustomWhiteCard } from '../../CustomMUIComponents/CustomCards';
 
 
 export default function EventColorSelector() {
     const navigate = useNavigate();
-    const [errorMessage, setErrorMessage] = useState('')
+    const [setErrorMessage] = React.useState('')
```
Comment (mnashat) [inline src/components/Calendar/EventColorSelector/EventColorSelector.js:21]: I'm not sure this implementation makes sense. If the state is never used, there is no point of having the stateSetter to begin with.

**Review by mnashat — COMMENTED** [review]
Good job. Please take a look at the comments.

File: src/components/StudyRoom/StudyRoomHome.js (hunk 1 above)
Comment (kikkiii21) [reply src/components/StudyRoom/StudyRoomHome.js:22]: yes

File: src/components/Calendar/EventColorSelector/EventColorSelector.js (hunk 2 above)
Comment (kikkiii21) [reply src/components/Calendar/EventColorSelector/EventColorSelector.js:21]: So im not sure we are using eventColorSelector in the code. The idea was to remove the SonarCloud issues first and then understand the code to see where exactly this is implemented and if we can get rid of it

Comment (kikkiii21) [reply src/components/Calendar/EventColorSelector/EventColorSelector.js:21]: the sonarCloud issue with this was the unused errorMessage

**Review by EyalAzimov — APPROVED** [review]

**Review by mnashat — APPROVED** [review]

**Pull request merged**